packages/core-commands/src/site-editor-navigation-commands.js

```diff
@@ -172,12 +172,16 @@ const getNavigationCommandLoaderPerTemplate = ( templateType ) =>
 				const extraArgs = isSiteEditor
 					? { canvas: getQueryArg( window.location.href, 'canvas' ) }
 					: {};
+				const templateLabel =
+					templateType === 'wp_template_part'
+						? __( 'template part' )
+						: __( 'template' );
 
 				return {
 					name: templateType + '-' + record.id,
 					searchLabel: record.title?.rendered + ' ' + record.id,
 					label: record.title?.rendered
-						? record.title?.rendered
+						? record.title?.rendered + ' ' + templateLabel
 						: __( '(no title)' ),
```

Comment:
What should we do if there is no title? However, in normal operation, I haven't been able to find a way to create templates or template parts without the title.